server/controllers/blogsController.js

```diff
@@ -1,100 +1,229 @@
     title: req.body.title,
     content: req.body.content,
     summary: req.body.summary,
     imageUrl: req.body.imageUrl,
-    minsRead:req.body.minsRead,
-    user_id: req.user.id,
-  })
-  // console.log(blog.user_id);
-
-  res.status(200).json(blog)
+    minsRead: req.body.minsRead,
+    author: req.user.id,
+  });
+
+  const blogWithAuthor = await Blog.aggregate([
+    { $match: { _id: blog._id } },
+    {
+      $lookup: {
+        from: "users",
+        localField: "author",
+        foreignField: "_id",
+        as: "authorInfo",
+      },
+    },
+    { $unwind: "$authorInfo" },
+    {
+      $addFields: {
+        authorName: "$authorInfo.fullName",
+      },
+    },
+    {
+      $project: {
+        authorInfo: 0,
+      },
+    },
+  ]);
+  res.status(200).json(blogWithAuthor[0]);
 });
 
-// Update a blog by ID
 exports.updateBlog = asyncHandler(async (req, res) => {
   const blogId = req.params.id;
   const { title, content, summary, imageUrl } = req.body;
 
   if (!title || !content || !summary || !imageUrl) {
     res.status(400);
-    throw new Error('Please fill all fields');
+    throw new Error("Please fill all fields");
   }
 
-  const updatedBlog = await Blog.findByIdAndUpdate(
-    blogId,
-    { title, content, summary, imageUrl },
-    { new: true } // Return the updated blog
-  );
+  const blog = await Blog.findById(blogId);
 
-  if (!updatedBlog) {
+  if (!blog) {
     res.status(404);
-    throw new Error('Blog not found');
+    throw new Error("Blog not found");
+  }
+
+  if (blog.author.toString() !== req.user.id) {
+    res.status(401);
+    throw new Error("Not authorized to update this blog");
   }
 
-  res.status(200).json(updatedBlog);
+  blog.title = title;
+  blog.content = content;
+  blog.summary = summary;
+  blog.imageUrl = imageUrl;
+
+  const updatedBlog = await blog.save();
+
+  const updatedBlogWithAuthor = await Blog.aggregate([
+    { $match: { _id: updatedBlog._id } },
+    {
+      $lookup: {
+        from: "users",
+        localField: "author",
+        foreignField: "_id",
+        as: "authorInfo",
+      },
+    },
+    { $unwind: "$authorInfo" },
+    {
+      $addFields: {
+        authorName: "$authorInfo.fullName",
+      },
+    },
+    {
+      $project: {
+        authorInfo: 0,
+      },
+    },
+  ]);
+  res.status(200).json(updatedBlogWithAuthor[0]);
 });
 
-// Delete a blog by ID
 exports.deleteBlog = asyncHandler(async (req, res) => {
   const blogId = req.params.id;
-  const deletedBlog = await Blog.findByIdAndRemove(blogId);
 
-  if (!deletedBlog) {
+  const blog = await Blog.findById(blogId);
+
+  if (!blog) {
     res.status(404);
-    throw new Error('Blog not found');
+    throw new Error("Blog not found");
   }
 
-  res.status(200).json({ message: 'Blog deleted successfully' });
+  if (blog.author.toString() !== req.user.id) {
+    res.status(401);
+    throw new Error("Not authorized to delete this blog");
+  }
+
+  await Blog.findByIdAndDelete(blogId);
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider adding error handling for deletion</b></div>
<div id="fix">

Consider adding error handling around the `findByIdAndDelete` operation in case deletion fails. The operation could fail due to network issues or database constraints. A similar issue was also found in client/src/pages/EditBlog.jsx (line 60-67).
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```suggestion

  try {
    const deletedBlog = await Blog.findByIdAndDelete(blogId);
    if (!deletedBlog) {
      res.status(500);
      throw new Error("Failed to delete blog");
    }
  } catch (error) {
    res.status(500);
    throw new Error("Error occurred while deleting blog: " + error.message);
  }
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/Divyanshu9822/blog-mern-app/pull/22#issuecomment-2544587479>#245782</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged